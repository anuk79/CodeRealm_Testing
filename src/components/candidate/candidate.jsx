// @flow

// vendors
import React from 'react';

const CandidateView = (props) => {
    return (
        <React.Fragment>
            <h3>Candidate Details:</h3>
            <div>
                <label>First Name:</label>
                <span>{props.candidateDetails.firstName}</span>
            </div>
            <div>
                <label>Last Name:</label>
                <span>{props.candidateDetails.lastName}</span>
            </div>
        </React.Fragment>
    );
}

export default CandidateView;