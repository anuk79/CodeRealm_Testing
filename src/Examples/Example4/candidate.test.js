
// vendors
import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// components
import { Candidate } from './candidate';

describe('Candidate component', () => {
    const mockFetchCandidateDetails = jest.fn();

    it('should call fetchCandidateDetails', () => {
        shallow(<Candidate 
            isDataFetching={true} 
            fetchCandidateDetails={mockFetchCandidateDetails}
        />);
        expect(mockFetchCandidateDetails).toBeCalled();
    });

    it('should render correctly when data is isDataFetching', () => {
        const wrapper = shallow(<Candidate 
            isDataFetching={true} 
            fetchCandidateDetails={mockFetchCandidateDetails}
        />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('LoadingSpinner').length).toBe(1);
    });

    it('should render correctly when some error', () => {
        const wrapper = shallow(<Candidate 
            hasError={true} 
            fetchCandidateDetails={mockFetchCandidateDetails}
        />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('.candidate-error').length).toBe(1);
    });

    it('should render correctly when no error and data isDataFetching is done', () => {
        const wrapper = mount(<Candidate 
            userDetails={{}} 
            fetchCandidateDetails={mockFetchCandidateDetails}
        />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('CandidateView').prop('userDetails')).toEqual({});
    });
});