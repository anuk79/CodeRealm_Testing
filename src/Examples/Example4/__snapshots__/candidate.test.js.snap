// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Candidate component should render correctly when data is isDataFetching 1`] = `
<LoadingSpinner
  customClassName="candidate-loading-spinner"
/>
`;

exports[`Candidate component should render correctly when no error and data isDataFetching is done 1`] = `
<Candidate
  candidateDetails={Object {}}
  fetchCandidateDetails={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
        Array [],
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
        Object {
          "isThrow": false,
          "value": undefined,
        },
        Object {
          "isThrow": false,
          "value": undefined,
        },
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  <CandidateView
    candidateDetails={Object {}}
  >
    <h3>
      Candidate Details:
    </h3>
    <div>
      <label>
        First Name:
      </label>
      <span />
    </div>
    <div>
      <label>
        Last Name:
      </label>
      <span />
    </div>
  </CandidateView>
</Candidate>
`;

exports[`Candidate component should render correctly when some error 1`] = `
<div
  className="candidate-error"
>
  Some error occurred. Please try again later.
</div>
`;
